### PR TITLE
Adding projects for the Bizkaia fiscalization + autogenerated Dtos

### PR DESCRIPTION
**Pull request by inakiesteve**
## Description

Adding autogenerated DTOS to support batuz fiscal integration. 

Fixes https://mews.atlassian.net/browse/OPS-1744

## Type of change
- [ ] Bug fix.
- [ x] Feature.
- [ ] Consolidation.

## Checklist
- [ ] Is breaking change.
- [ ] Documentation updated.
- [ ] Tests included (please specify cases).

A few other PRs will follow this one with models and unit tests to validate the DTOs.

### CODE REVIEW

File: src/Mews.Fiscalizations.All.sln [hunk 1]
```diff
@@ -52,6 +52,12 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Mews.Fiscalizations.Basque"
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Mews.Fiscalizations.Basque.Tests", "Basque\Mews.Fiscalizations.Basque.Tests\Mews.Fiscalizations.Basque.Tests.csproj", "{9A27DA76-278F-4B87-8AE3-0504C09AA1B8}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Bizkaia", "Bizkaia", "{B33E5A1D-A667-477D-9AD8-1673192B71BB}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Mews.Fiscalizations.Bizkaia", "Mews.Fiscalizations.Bizkaia\Mews.Fiscalizations.Bizkaia.csproj", "{145C287C-2D14-4B35-A73A-6A97ABBF3F10}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Mews.Fiscalizations.Bizkaia.Tests", "Mews.Fiscalizations.Bizkaia.Tests\Mews.Fiscalizations.Bizkaia.Tests.csproj", "{2767CE08-C4C2-4043-AC58-C413ECB64D3A}"
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalizations.All.sln:59]: If i remember correctly, Bizkaia and the other 2 regions share most of the stuff (almost identical XSDs)

so i suggest that we either implement Bizkaia within the same project (Basque), or rename the Basque project to something else like GipuzkoaAndAraba, or having a project per region.

I would go with the first solution (all regions within 1 project).

File: src/Mews.Fiscalizations.Bizkaia.Tests/Mews.Fiscalizations.Bizkaia.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.1.0" />
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalizations.Bizkaia.Tests/Mews.Fiscalizations.Bizkaia.Tests.csproj:12]: There should be a new version of this package, pls sync it with the rest of the projects.

File: src/Mews.Fiscalizations.Bizkaia.Tests/Mews.Fiscalizations.Bizkaia.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    <PackageReference Include="NUnit" Version="3.13.3" />
+    <PackageReference Include="NUnit3TestAdapter" Version="4.2.1" />
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalizations.Bizkaia.Tests/Mews.Fiscalizations.Bizkaia.Tests.csproj:14]: same here

File: src/Mews.Fiscalizations.Bizkaia.Tests/Mews.Fiscalizations.Bizkaia.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    <PackageReference Include="NUnit.Analyzers" Version="3.3.0" />
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalizations.Bizkaia.Tests/Mews.Fiscalizations.Bizkaia.Tests.csproj:15]: mmm nice one, i didn't know about it before, perhaps we should use it for the other test projects

File: src/Mews.Fiscalizations.Bizkaia.Tests/Mews.Fiscalizations.Bizkaia.Tests.csproj [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    <PackageReference Include="coverlet.collector" Version="3.1.2" />
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalizations.Bizkaia.Tests/Mews.Fiscalizations.Bizkaia.Tests.csproj:16]: I don't see any usage of this lib

File: src/Mews.Fiscalizations.Bizkaia/Mews.Fiscalizations.Bizkaia.csproj [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Folder Include="Dto\" />
+  </ItemGroup>
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalizations.Bizkaia/Mews.Fiscalizations.Bizkaia.csproj:11]: You can delete this

File: src/Mews.Fiscalizations.Bizkaia.Tests/Mews.Fiscalizations.Bizkaia.Tests.csproj [hunk 7]
```diff
@@ -0,0 +1,19 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalizations.Bizkaia.Tests/Mews.Fiscalizations.Bizkaia.Tests.csproj:7]: Remove the extra whitespace

File: src/Mews.Fiscalizations.Bizkaia/Mews.Fiscalizations.Bizkaia.csproj
Comment (abdallahbeshi) [inline src/Mews.Fiscalizations.Bizkaia/Mews.Fiscalizations.Bizkaia.csproj:1]: If you decide to keep this as a separate project, then there are many properties missing here, like the License/PackageVersion....etc please check the other projects.

File: src/Mews.Fiscalizations.Bizkaia/Dto/BatuzInvoiceRequest.cs
Comment (abdallahbeshi) [inline src/Mews.Fiscalizations.Bizkaia/Dto/BatuzInvoiceRequest.cs:1]: What's the difference between `BatuzInvoiceRequest` and `InvoiceRequest`?

I believe there is a LROE request and TicketBai request (and TicketBai invoice request will be sent as a part of the LROE request)

File: src/Mews.Fiscalizations.Bizkaia/Dto/BatuzInvoiceResponse.cs [hunk 8]
```diff
@@ -0,0 +1,489 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+using static System.Net.WebRequestMethods;
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalizations.Bizkaia/Dto/BatuzInvoiceResponse.cs:7]: The only used using here is `using System.Xml.Serialization;`

File: src/Mews.Fiscalizations.Bizkaia/Dto/BatuzInvoiceResponse.cs [hunk 9]
```diff
@@ -0,0 +1,489 @@
...
+
+namespace Mews.Fiscalizations.Bizkaia.Dto
+{
```
Comment (abdallahbeshi) [inline src/Mews.Fiscalizations.Bizkaia/Dto/BatuzInvoiceResponse.cs:10]: Please convert to file-scoped namespace.

**Review by abdallahbeshi — CHANGES_REQUESTED** [review]

File: src/Mews.Fiscalizations.Bizkaia.Tests/Mews.Fiscalizations.Bizkaia.Tests.csproj (hunk 4 above)
Comment (abdallahbeshi) [reply src/Mews.Fiscalizations.Bizkaia.Tests/Mews.Fiscalizations.Bizkaia.Tests.csproj:15]: Or maybe not 😄, just noticed that this is not an official library of NUnit, so we would need to do some research before we start using it.

File: src/Mews.Fiscalizations.All.sln (hunk 1 above)
Comment (inakiesteve) [reply src/Mews.Fiscalizations.All.sln]: Ok, I will go with option 1 and have all regions in the same project..

File: src/Mews.Fiscalizations.Bizkaia/Dto/BatuzInvoiceRequest.cs
Comment (inakiesteve) [reply src/Mews.Fiscalizations.Bizkaia/Dto/BatuzInvoiceRequest.cs:1]: So Invoice request is basically what allows you to create a TicketBai invoice.

BatuzRequest is what allows you to send a request to the Batuz system or a LROE request like you mentioned (I could rename this to LROERequest if we feel it is a better name for this purpose). When we send a request of this kind to batuz we should at least include one ticketBai invoice encoded in Base64.

File: src/Basque/Mews.Fiscalizations.Basque/Dto/Bizkaia/LROERequest.cs [hunk 10]
```diff
@@ -0,0 +1,496 @@
...
+
+// NOTE: Generated code may require at least .NET Framework 4.5 or .NET Core/Standard 2.0.
+/// <remarks/>
+[Serializable()]
+[System.ComponentModel.DesignerCategory("code")]
+[XmlType(AnonymousType = true, Namespace = "https://www.batuz.eus/fitxategiak/batuz/LROE/esquemas/LROE_PJ_240_1_1_FacturasEmi" +
+    "tidas_ConSG_AltaPeticion_V1_0_2.xsd")]
+[XmlRoot(Namespace = "https://www.batuz.eus/fitxategiak/batuz/LROE/esquemas/LROE_PJ_240_1_1_FacturasEmi" +
+    "tidas_ConSG_AltaPeticion_V1_0_2.xsd", IsNullable = false)]
+public class LROEPJ240FacturasEmitidasConSGAltaPeticion
```
Comment (abdallahbeshi) [inline src/Basque/Mews.Fiscalizations.Basque/Dto/Bizkaia/LROERequest.cs:14]: Let's confirm with the research team that only model 240 is applicable in our case.

**Review by abdallahbeshi — CHANGES_REQUESTED** [review]

**Review by abdallahbeshi — APPROVED** [review]

**Pull request merged**